crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -2121,6 +2134,737 @@ mod tests {
+                    invocation_id,
+                    invocation_target: invocation_target.clone(),
+                    ..ServiceInvocation::mock()
+                }))
+                .await;
+
+            let _ = state_machine
+                .apply(Command::Invoke(ServiceInvocation {
+                    invocation_id: inboxed_id,
+                    invocation_target: inboxed_target,
+                    response_sink: Some(ServiceInvocationResponseSink::PartitionProcessor {
+                        caller: caller_id,
+                        entry_index: 0,
+                    }),
+                    ..ServiceInvocation::mock()
+                }))
+                .await;
+
+            let current_invocation_status = state_machine
+                .storage()
+                .transaction()
+                .get_invocation_status(&inboxed_id)
+                .await?;
+
+            // assert that inboxed invocation is in invocation_status
+            assert!(let InvocationStatus::Inboxed(_) = current_invocation_status);
+
+            let actions = state_machine
+                .apply(Command::TerminateInvocation(InvocationTermination::kill(
+                    inboxed_id,
+                )))
+                .await;
+
+            let current_invocation_status = state_machine
+                .storage()
+                .transaction()
+                .get_invocation_status(&inboxed_id)
+                .await?;
+
+            // assert that invocation status was removed
+            assert!(let InvocationStatus::Free = current_invocation_status);
+
+            fn outbox_message_matcher(
+                caller_id: InvocationId,
+            ) -> impl Matcher<ActualT = restate_storage_api::outbox_table::OutboxMessage>
+            {
+                pat!(
+                    restate_storage_api::outbox_table::OutboxMessage::ServiceResponse(pat!(
+                        restate_types::invocation::InvocationResponse {
+                            id: eq(caller_id),
+                            entry_index: eq(0),
+                            result: eq(ResponseResult::Failure(KILLED_INVOCATION_ERROR))
+                        }
+                    ))
+                )
+            }
+
+            assert_that!(
+                actions,
+                contains(pat!(Action::NewOutboxMessage {
+                    message: outbox_message_matcher(caller_id)
+                }))
+            );
+
+            let partition_id = state_machine.partition_id();
+            let outbox_message = state_machine
+                .storage()
+                .transaction()
+                .get_next_outbox_message(partition_id, 0)
+                .await?;
+
+            assert_that!(
+                outbox_message,
+                some((ge(0), outbox_message_matcher(caller_id)))
+            );
+
+            Ok(())
+        }
+
+        #[test(tokio::test(flavor = "multi_thread", worker_threads = 2))]
```

Comment:
Is there a specific reason why the number of worker threads is 2 and we are not using the current thread runtime?

Reply:
IDK about this, this change was brought in by @AhmedSoliman, I'm simply following the way the other tests were changed.

Reply:
Not needed anymore. There was a time during a transition when that was required. Feel free to change it back if needed.